tools/json2cbor/json2cbor.c

```diff
@@ -154,7 +154,8 @@ uint8_t *decode_base16(const char *string, size_t *len)
     if (buffer == NULL)
         return NULL;
 
-    for (size_t i = 0; i < *len; ++i) {
+    size_t i;
```

Comment:
Well, mixing variable declarations and statements is also only allowed in C99. If you're going to try and solve the warning, move the declaration to the beginning of the function.